app/javascript/components/supplemental_application/actions.js

```diff
@@ -8,20 +8,20 @@ export const updateApplication = async (application, prevApplication) => {
   const promises = [updateLease(application['lease'], application['primaryApplicantContact'], application['id'])]
 
   // Conctact lease promise with rental assistances
-  promises.concat(updateRentalAssistances(application, prevApplication))
+  promises.concat(updateUnsavedRentalAssistances(application, prevApplication))
   const applicationApi = domainToApi.buildApplicationShape(application)
   promises.push(apiService.submitApplication(applicationApi))
   const responses = await Promise.all(promises)
 
   return every(responses, (promise) => promise !== false)
 }
 
-const updateRentalAssistances = (application, prevApplication) => {
+const updateUnsavedRentalAssistances = (application, prevApplication) => {
```

Comment:
This function is different from `handleSaveRentalAssistance` because `updateUnsavedRentalAssistances` is finding _and_ saving the unsaved rental assistances, while `handleSaveRentalAssistance` directly creates/updates a single rental assistance. I agree with this approach and just renamed it to clarify the uniqueness.